File: csharp/sprint2/D/Solution.cs
#if !REMOTE_JUDGE
public class Node<TValue>
{
    public TValue Value { get; private set; }
    public Node<TValue> Next { get; set; }

    public Node(TValue value, Node<TValue> next)
    {
        Value = value;
        Next = next;
    }
}
#endif

public class Solution
{
    public static int Solve(Node<string> head, string item)
    {
        // Your code
        // ヽ(´▽`)/
    }

    #if !REMOTE_JUDGE
    private static void Main()
    {
        var node3 = new Node<string>("node3", null);
        var node2 = new Node<string>("node2", node3);
        var node1 = new Node<string>("node1", node2);
        var node0 = new Node<string>("node0", node1);
        var idx = Solution.Solve(node0, "node2");
        // result is : idx == 2
    }
    #endif
}

File: swift/sprint2/D/solution.swift
//   Comment it before submitting

//   open class Node {
//
//   public var value: String
//   public var next: Node?
//
//   public init(_ value: String) {
//     self.value = value
//     self.next = nil
//   }
// }
 
func solution(head: Node?, elem: String) -> Int {
    // Your code
    // ヽ(´▽`)/
    return -1;
}


func test() {
    let node3 = Node("node3")
    let node2 = Node("node2")
    node2.next = node3
    let node1 = Node("node1")
    node1.next = node2
    let node0 = Node("node0")
    node0.next = node1
    let idx = solution(head: node0, elem: "node2")
    assert(idx == 2)
}